themes/qgis/layouts/partials/menu.html

```diff
@@ -35,18 +35,26 @@
                 {{ end }}
             {{ end }}
             <div class="navbar-item">
-                <div class="control has-icons-right mr-2">
+                <form action="/search" method="GET" id="search">
```

Comment:
All the changes in this file would be better done by keeping the original implementation and just adding the id, search button to the original. i.e. make the minimal changes that are needed to achieve your implementation rather than just copying everything verbatim from the example you based your work on.